src/coreclr/jit/optimizer.cpp

```diff
@@ -4575,6 +4575,267 @@ PhaseStatus Compiler::optUnrollLoops()
+}
+
+bool Compiler::optIfConvertSelect(GenTree* original_condition, GenTree* asg_node)
+{
+    assert(original_condition->OperIsCmpCompare());
+    assert(asg_node->gtOper == GT_ASG);
+
+    // Duplicate the input of the assign.
+    // This will be used as the false result of the select node.
+    GenTree* current_value = gtCloneExpr(asg_node->AsOp()->gtOp1);
+    current_value->gtFlags &= GTF_EMPTY;
+
+    // Duplicate the condition and invert it
+    GenTree* cond = gtCloneExpr(original_condition);
+    cond->gtFlags |= GTF_DONT_CSE;
+    cond->gtFlags ^= GTF_RELOP_JMP_USED;
+    cond->gtFlags ^= GTF_RELOP_NAN_UN;
+    cond->gtOper = GenTree::ReverseRelop(cond->gtOper);
+
+    // Create a select node.
+    GenTreeConditional* select =
+        gtNewConditionalNode(GT_SELECT, cond, asg_node->AsOp()->gtOp2, current_value, asg_node->TypeGet());
+
+    // Use the select as the input to the assignment.
+    asg_node->AsOp()->gtOp2 = select;
+    asg_node->AsOp()->gtFlags |= (select->gtFlags & GTF_ALL_EFFECT);
+
+    // Calculate costs.
+    gtSetEvalOrder(select);
+
+    return true;
+}
+
+bool Compiler::optIfConvertCCmp(GenTree* original_condition, GenTree* asg_node)
+{
+    assert(original_condition->OperIsCmpCompare());
+    assert(asg_node->gtOper == GT_ASG);
+
+    // For now, don't handle floats.
+    if (!varTypeIsIntegralOrI(original_condition->AsOp()->gtOp1->TypeGet()))
+    {
+        return false;
+    }
+
+    // Note: Limiting the maximum number of chained compares may give a performance
+    //       boost, at the cost of more emitted code.
+
+    GenTreeConditional* select_node      = asg_node->AsOp()->gtOp2->AsConditional();
+    GenTree*            select_node_cond = select_node->gtCond;
+
+    // Duplicate the condition and invert it
+    GenTree* new_cond = gtCloneExpr(original_condition);
+    new_cond->gtFlags |= GTF_DONT_CSE;
+    new_cond->gtFlags ^= GTF_RELOP_JMP_USED;
+    new_cond->gtFlags ^= GTF_RELOP_NAN_UN;
+    new_cond->gtOper = GenTree::ReverseRelop(new_cond->gtOper);
+
+    // Link together the two conditions.
+    GenTree* const link = gtNewOperNode(GT_AND, TYP_INT, select_node_cond, new_cond);
+    link->gtFlags |= (select_node_cond->gtFlags & GTF_ALL_EFFECT);
+    link->gtFlags |= (new_cond->gtFlags & GTF_ALL_EFFECT);
+
+    // Use the link as the conditional input to the select.
+    select_node->gtCond = link;
+    select_node->gtFlags |= (link->gtFlags & GTF_ALL_EFFECT);
+    select_node->gtFlags ^= GTF_RELOP_NAN_UN;
+    asg_node->AsOp()->gtFlags |= (select_node->gtFlags & GTF_ALL_EFFECT);
+
+    // Calculate costs.
+    gtSetEvalOrder(link);
+
+    return true;
+}
+
+//-----------------------------------------------------------------------------
+// optIfConversion: If conversion
+//
+// Returns:
+//   suitable phase status
+//
+void Compiler::optIfConversion()
+{
+    // Reverse iterate through the blocks.
+    BasicBlock* block = fgLastBB;
+    while (block != nullptr)
+    {
+        optIfConvert(block);
```

Comment:
You are not actually returning a suitable phase status.

Consider making `optIfConvert` return true/false if any conversions happened and update this method accordingly.

Reply:
>You are not actually returning a suitable phase status.

My original version had this, but when I moved the phase to after assertion prop, I noticed the other phases around here didn't either:
`void       optAssertionPropMain();`
`void optOptimizeCSEs();`
So, I removed it to match them.

Reply:
> I noticed the other phases around here didn't either

The will soon...

Reply:
> The will soon...

:) Ok, I'll update `optIfConvert`